splunk/resource_splunk_data_ui_views.go

```diff
@@ -43,24 +43,15 @@ func splunkDashboardsCreate(d *schema.ResourceData, meta interface{}) error {
 	provider := meta.(*SplunkProvider)
 	name := d.Get("name").(string)
 	splunkDashboardsObj := getSplunkDashboardsConfig(d)
-	aclObject := &models.ACLObject{}
-	if r, ok := d.GetOk("acl"); ok {
-		aclObject = getACLConfig(r.([]interface{}))
-	} else {
-		aclObject.App = "search"
-		aclObject.Owner = "admin"
-		aclObject.Sharing = "user"
-	}
+	aclObject := getResourceDataViewACL(d)
+
 	err := (*provider.Client).CreateDashboardObject(aclObject.Owner, aclObject.App, splunkDashboardsObj)
 	if err != nil {
 		return err
 	}
 
-	if _, ok := d.GetOk("acl"); ok {
-		err = (*provider.Client).UpdateAcl(aclObject.Owner, aclObject.App, name, aclObject, "data", "ui", "views")
-		if err != nil {
-			return err
-		}
+	if err := (*provider.Client).UpdateAcl(aclObject.Owner, aclObject.App, name, aclObject, "data", "ui", "views"); err != nil {
+		return err
```

Comment:
Since I had to make a change to these lines I also changed it to scope `err` to the `if` block.